libevm/VM.cpp

```diff
@@ -653,6 +660,10 @@ void VM::interpretCases()
 
 		CASE(SAR)
 		{
+			// Pre-constantinople
+			if (!m_schedule->haveBitwiseShifting)
+				throwBadInstruction();
+
 			ON_OP();
 			updateIOGas();
 
```

Comment:
There is a missed else-endif block below.

Reply:
Thanks Alex :)